### DISCUSSION TITLE
Long Delay for User Consent in Meshcentral Connections

### DISCUSSION THREAD

**Discussion by Domkra** [Ylianst/MeshCentral#6368] [Issues]
Hello everyone,

I have a question. Recently, I noticed that when I try to connect to a desktop using Meshcentral, the request for user consent takes about 90 to 120 seconds before the customer can confirm access.

However, when I use Meshcentral without the consent request, the connection to the desktop is established in 1 second.

Is this normal behavior or a possible error? Are there ways to understand the reason for the long delay? Or can this behavior be caused by the configuration and be turned off?

Best regards

**Comment by si458** [comment]
its hard to say as it could be your internet/remote devices/meshcentral setup

but it could also be a bug with the user-consent section in meshcore.js as somebody pointed out that the consent box was crashing/hanging on certain windows devices

one thing you can try to fix it, it to restart the meshagent service
(`service restart` in the console tab of the device OR restart service on the device itself OR restart the whole device)

**Comment by Domkra** [comment]
I believe the issue is not related to the internet or the remote devices, as accessing the remote desktop without user consent works without any problems.

----------------
Here is what I have tried so far:
- Restarting the MeshAgent service (Console tab) on multiple devices = No success
- Restarting the MeshAgent service (on the device) = No success
- Restarting the entire device, server, and service = No success
- Updating MeshAgent = No success

----------------
We are using MeshCentral version 1.1.29 with Node v20.17.0.

**Reply by si458** [reply]
Open up a bug report then, follow the template, include all the info and I'll try have a look when I get chance as I cannot replicate that type of behaviour here

Only thing I can think of is the physical machine itself with a graphics driver issue, make sure ur drivers are all up to date?

**Comment by Domkra** [comment]
Thank you for your efforts. I will create a bug report.

The problem occurs on all devices from our IT, and the graphics drivers are up to date, so I don't think that's the issue.

**Reply by si458** [reply]
just out of interest are you having this same here https://github.com/Ylianst/MeshCentral/issues/6290
where on some machines the consent box like crashes?

**Reply by Domkra** [reply]
“No, for us it just takes a very long time for the consent window to appear. After clicking it, I can provide normal remote support. The problem you mentioned has not occurred for us yet.”

**Reply by si458** [reply]
hmmm ok no worries just wanted to check 👍
